docker-compose.yaml

```diff
@@ -0,0 +1,306 @@
+  zoekt-indexserver-0:
+    container_name: zoekt-indexserver-0
+    image: >-
+      sourcegraph/zoekt-indexserver:0.0.20191031121923-5bd7e84@sha256:43eac5fddcc32f84f097e1466918ec9ac2fe54fe34649b8399de93963d9e4916
+    environment:
+      - GOMAXPROCS=8
+      - 'HOSTNAME=zoekt-webserver-0:6070'
+      - 'SRC_FRONTEND_INTERNAL=http://sourcegraph-frontend-internal:3090'
+    volumes:
+      - '~/sourcegraph-docker/zoekt-0-shared-disk:/data/index'
+    networks:
+      - sourcegraph
+    restart: always
+    hostname: zoekt-indexserver-0
+  zoekt-webserver-0:
+    container_name: zoekt-webserver-0
+    image: >-
+      sourcegraph/zoekt-webserver:0.0.20191031121751-5bd7e84@sha256:1afbfb746a3d43a7532693093d2f993dac5b5536037b31d3ab305ceef5687380
+    environment:
+      - GOMAXPROCS=16
+      - 'HOSTNAME=zoekt-webserver-0:6070'
+    volumes:
+      - '~/sourcegraph-docker/zoekt-0-shared-disk:/data/index'
+    networks:
+      - sourcegraph
+    restart: always
+    hostname: zoekt-webserver-0
+  sourcegraph-frontend-internal:
+    container_name: sourcegraph-frontend-internal
+    image: >-
+      sourcegraph/frontend:3.10.4@sha256:5791fb16a1a2828f14c227160104eba3a251c876359e0cb33c48f832e3d2a53f
+    environment:
+      - GOMAXPROCS=4
+      - PGHOST=pgsql
+      - 'SRC_GIT_SERVERS=gitserver-0:3178'
+      - 'SRC_SYNTECT_SERVER=http://syntect-server:9238'
+      - 'SEARCHER_URL=http://searcher-0:3181'
+      - 'SYMBOLS_URL=http://symbols-0:3184'
+      - 'INDEXED_SEARCH_SERVERS=zoekt-webserver-0:6070'
+      - 'SRC_FRONTEND_INTERNAL=sourcegraph-frontend-internal:3090'
+      - 'REPO_UPDATER_URL=http://repo-updater:3182'
+      - 'REPLACER_URL=http://replacer:3185'
+      - 'LSIF_SERVER_URL=http://lsif-server:3186'
+      - 'GRAFANA_SERVER_URL=http://grafana:3000'
+    volumes:
+      - >-
+        ~/sourcegraph-docker/sourcegraph-frontend-internal-0-disk:/mnt/cache
+    networks:
+      - sourcegraph
+    restart: always
+  sourcegraph-frontend-2:
+    container_name: sourcegraph-frontend-2
+    image: >-
+      sourcegraph/frontend:3.10.4@sha256:5791fb16a1a2828f14c227160104eba3a251c876359e0cb33c48f832e3d2a53f
+    environment:
+      - GOMAXPROCS=12
+      - JAEGER_AGENT_HOST=jaeger-agent
+      - PGHOST=pgsql
+      - 'SRC_GIT_SERVERS=gitserver-0:3178'
+      - 'SRC_SYNTECT_SERVER=http://syntect-server:9238'
+      - 'SEARCHER_URL=http://searcher-0:3181'
+      - 'SYMBOLS_URL=http://symbols-0:3184'
+      - 'INDEXED_SEARCH_SERVERS=zoekt-webserver-0:6070'
+      - 'SRC_FRONTEND_INTERNAL=sourcegraph-frontend-internal:3090'
+      - 'REPO_UPDATER_URL=http://repo-updater:3182'
+      - 'REPLACER_URL=http://replacer:3185'
+      - 'LSIF_SERVER_URL=http://lsif-server:3186'
+      - 'GRAFANA_SERVER_URL=http://grafana:3370'
+    volumes:
+      - >-
+        ~/sourcegraph-docker/sourcegraph-frontend-2-disk:/mnt/cache
+    ports:
+      - '0.0.0.0:3082:3080'
+    networks:
+      - sourcegraph
+    restart: always
+  sourcegraph-frontend-1:
```

Comment:
The docs we have in the scripts are rather helpful in trying to read these things -- as it stands this compose file is pretty illegible. We can just copy those docs into here.

Reply:
Copied and re-ordered related services.